packages/ts/react-grid/src/autogrid.tsx

```diff
@@ -98,9 +111,11 @@ export function AutoGrid<TItem>({ service, model, filter, ...gridProps }: AutoGr
 
   useEffect(() => {
     // Update the filtering, whenever the filter changes
-    const grid = ref.current!;
-    dataProviderFilter.current = filter;
-    grid.clearCache();
+    const grid = ref.current;
+    if (grid) {
+      dataProviderFilter.current = filter;
+      grid.clearCache();
+    }
```

Comment:
All tests fail for me without this check

```
ERROR LOG: 'Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.'

  @hilla/react-grid
    useAutoGrid
      ✗ creates columns based on model
        Error: Uncaught TypeError: Cannot read properties of null (reading 'useRef') (http://localhost:9877/packages/ts/react-grid/.vite/deps/chunk-U33FUF2H.js?v=c646efa1:1074)
            at Object.invokeGuardedCallbackDev (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:3699:24)
            at invokeGuardedCallback (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:3733:39)
            at beginWork$1 (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:19761:15)
            at performUnitOfWork (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:19197:20)
            at workLoopSync (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:19133:13)
            at renderRootSync (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:19112:15)
            at performConcurrentWorkOnRoot (packages/ts/react-grid/.vite/deps/chunk-HMFLAOO4.js?v=a669691a:18674:83)
            at flushActQueue (packages/ts/react-grid/.vite/deps/chunk-DFDKWVIJ.js?v=a669691a:1812:32)
            at act (packages/ts/react-grid/.vite/deps/chunk-DFDKWVIJ.js?v=a669691a:1747:19)
```
Hook calls in the component look fine though 🤔

Reply:
Not sure how you run the tests because after removing the if, I get
```
  @hilla/react-grid
    useAutoGrid
      ✓ creates columns based on model
      ✓ sets a data provider, but only once
      ✓ data provider provides data
      ✓ does not pass its own parameters to the underlying grid
      ✓ passes filter to the data provider
      customize columns
        ✓ should only show configured columns in specified order
        ✓ should ignore unknown columns
```